Pre-DevIncubator/Vehicle.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pre_DevIncubator
+{
+    public class Vehicle : IComparable<Vehicle>
+    {
+        private readonly VehicleType vehicleType;
+        private readonly string modelName;
+        private string registrationNumber;
+        private int weight;
+        private readonly int manufactureYear;
+        private int mileage;
+        private Color color;
+
+        public VehicleType VehicleType => vehicleType;
+
+        public string ModelName => modelName;
+
+        public string RegistrationNumber { get => registrationNumber; set => registrationNumber = value; }
+        public int Weight { get => weight; set => weight = value; }
+
+        public int ManufactureYear => manufactureYear;
+
+        public int Mileage { get => mileage; set => mileage = value; }
+        public Color Color { get => color; set => color = value; }
```

Comment:
Why do you use Properties with fields instead AutoProperties? This approach of creating properties with fields will be usefull if we have to validate our data.
```suggestion
       
        public VehicleType VehicleType {get; init;}
        public string ModelName {get; init;};
        public string RegistrationNumber { get; set; }
        public int Weight { get; set; }
        public int ManufactureYear {get; init;}
        public int Mileage { get; set; }
        public Color Color { get; set; }
```